applications/GeoMechanicsApplication/python_scripts/set_parameter_field_process.py

```diff
@@ -0,0 +1,111 @@
+    script. This python script has to be inherited from the 'ParameterFieldBase' class. Which is located at:
+    'GeoMechanicsApplication->python_scripts->user_defined_scripts->user_defined_parameter_field_base.py'
+    the name of the script (without '.py') should be filled in at the 'function' parameter in the projectparameters.json
+
+    | option 3, func_type = 'json': with this option, a parameter field can be directly read from a json dictionary.
+    This dictionary has to contain the 'values' key, which is a 1D list of all the field values. The list has to have
+    the same size as the elements within the model part, and need to be accordingly sorted. The filename should be
+    filled in at the 'dataset' parameter, within the projectparameters.json
+    """
+
+    def __init__(self, Model, settings ):
+        KratosMultiphysics.Process.__init__(self)
+
+        self.model_part = Model[settings["model_part_name"].GetString()]
+
+        self.params = KratosMultiphysics.Parameters("{}")
+        self.params.AddValue("model_part_name",settings["model_part_name"])
+        self.params.AddValue("variable_name", settings["variable_name"])
+        self.params.AddValue("func_type", settings["func_type"])
+        self.params.AddValue("function", settings["function"])
+        self.params.AddValue("dataset", settings["dataset"])
+        self.process = KratosGeo.SetParameterFieldProcess(self.model_part, self.params)
+
+    def ExecuteInitialize(self):
+        """
+        Initializes the process. Within the python part of 'ExecuteInitialize', the parameter field with func type
+        'python' is generate. The cpp part sets the fields on the elements.
+
+        Returns
+        -------
+
+        """
+
+        # if the type of parameter field is a user defined python script:
+        if self.params["func_type"].GetString() == "python":
+
+            # initialise input and output
+            input_dict = {}
+            return_dict = {}
+
+            values = []
+            all_coordinates = []
+
+            # Get variable object
+            imported_modules = [KratosGeo, KratosMultiphysics]
+
+            variable = None
+            for kratos_module in imported_modules:
+                if hasattr(kratos_module, self.params["variable_name"].GetString()):
+                    variable = getattr(kratos_module, self.params["variable_name"].GetString())
+                    break
+
+            # raise an error if the variable is not imported
+            if variable is None:
+                raise AttributeError(f'The variable: {self.params["variable_name"].GetString()} is not present within the '
+                                     f'imported modules')
+
+            for element in self.model_part.Elements:
+
+                # calculate center coordinates of the element
+                center = element.GetGeometry().Center()
+                coords = [center[0], center[1], center[2]]
```

Comment:
What is the difference between center and coords, they both appear to be lists of the x,y,z ???